src/main.rs

```diff
@@ -12,27 +13,72 @@ use lightswitch::object::build_id;
 use lightswitch::profiler::Collector;
 use lightswitch::profiler::Profiler;
 use lightswitch::unwind_info::{compact_printing_callback, UnwindInfoBuilder};
+use primal::is_prime;
 use std::error::Error;
 use std::path::PathBuf;
 
 use std::time::Duration;
 
+fn parse_duration(arg: &str) -> Result<Duration, std::num::ParseIntError> {
+    let seconds = arg.parse()?;
+    Ok(Duration::from_secs(seconds))
+}
+
+const SAMPLE_FREQ_RANGE: RangeInclusive<usize> = 1..=1001;
+
+fn sample_freq_in_range(s: &str) -> Result<u16, String> {
+    let sample_freq: usize = s
+        .parse()
+        .map_err(|_| format!("`{s}' isn't a valid frequency"))?;
+    if !(1..=1001).contains(&sample_freq) {
```

Comment:
I assume you wanted to use the constant here?
```suggestion
    if !SAMPLE_FREQ_RANGE.contains(&sample_freq) {
```

Reply:
Ugh - yes, fixed.